internal/operators/odf/validations.go

```diff
@@ -68,28 +68,29 @@ func (o *operator) computeResourcesAllNodes(cluster *models.Cluster, odfClusterR
 	var err error
 
 	hosts := cluster.Hosts
-	compactMode := int64(len(hosts)) == 3
+	isCompactMode := int64(len(hosts)) == 3
 	for _, host := range hosts { // if the worker nodes >=3 , install ODF on all the worker nodes if they satisfy ODF requirements
 
 		/* If the Role is set to Auto-assign for a host, it is not possible to determine whether the node will end up as a master or worker node.
 		As ODF will use only worker nodes for non-compact deployments, the ODF validations cannot be performed as it cannot know which nodes will be worker nodes.
 		We ignore the role check for a cluster of 3 nodes as they will all be master nodes. ODF validations will proceed as for a compact deployment.
 		*/
+
 		role := common.GetEffectiveRole(host)
-		if !compactMode {
+		if !isCompactMode {
 			if role == models.HostRoleAutoAssign {
 				status = "For ODF Standard Mode, all host roles must be assigned to master or worker."
 				err = errors.New("Role is set to auto-assign for host ")
 				return status, err
 			}
 			if role == models.HostRoleWorker {
-				status, err = o.computeNodeResourceUtil(host, odfClusterResources)
+				status, err = o.computeNodeResourceUtil(host, odfClusterResources, "standard")
 				if err != nil {
 					return status, err
 				}
 			}
 		} else {
-			status, err = o.computeNodeResourceUtil(host, odfClusterResources)
+			status, err = o.computeNodeResourceUtil(host, odfClusterResources, "compact")
```

Comment:
`"compact"` is not a valid value for a `odfDeploymentMode`, please use the constants